core/core.go

```diff
@@ -61,6 +61,9 @@ type IpfsNode struct {
 	// Self
 	Identity peer.ID // the local node's identity
 
+	// FIXME: Check if this is the right place for it.
+	UserAgentSuffix string
```

Comment:
Perhaps we could move this to `version.go` and have `SetUserAgentSuffix` for updating the suffix  at the runtime when the CLI flag is present and `GetUserAgentVersion()` for reading the entire thing, instead of passing the suffix each time?
This way we would have everything in one place, hidden away.

Reply:
Take a look at this [comment](https://github.com/ipfs/go-ipfs/pull/8419/files#diff-1ef170619a70876f007d5edfc4554a81aa686eae7678b70df0347b3133cd6d14R18-R22) please and let me know if it's aligned with the request here, just for me to make sure I understand it correctly.

Basically the suggestion is having a _single, static_ suffix for the entire program and allowing modifications on the fly (that would ideally be contained in the `ipfs daemon` startup through the setter you're suggesting).

Reply:
Yes, the alternative solution described in the comment sounds preferable :+1: 
(it will make things easier if we ever decide that we want to allow an alternative way of defining the suffix by defining it in the config file)

Reply:
Sounds good, thanks for the confirmation.